src/sagemaker_xgboost_container/mms_patch/model_server.py

```diff
@@ -0,0 +1,178 @@
+        os.makedirs(DEFAULT_MMS_MODEL_DIRECTORY)
+
+    model_archiver_cmd = ['model-archiver',
+                          '--model-name', DEFAULT_MMS_MODEL_NAME,
+                          '--handler', handler_service,
+                          '--model-path', environment.model_dir,
+                          '--export-path', DEFAULT_MMS_MODEL_DIRECTORY,
+                          '--archive-format', 'no-archive',
+                          ]
+
+    logger.info(model_archiver_cmd)
+    subprocess.check_call(model_archiver_cmd)
+
+
+def _set_python_path():
+    # MMS handles code execution by appending the export path, provided
+    # to the model archiver, to the PYTHONPATH env var.
+    # The code_dir has to be added to the PYTHONPATH otherwise the
+    # user provided module can not be imported properly.
+    code_dir_path = '{}:'.format(environment.code_dir)
+
+    if PYTHON_PATH_ENV in os.environ:
+        os.environ[PYTHON_PATH_ENV] = code_dir_path + os.environ[PYTHON_PATH_ENV]
+    else:
+        os.environ[PYTHON_PATH_ENV] = code_dir_path
+
+
+def _create_model_server_config_file():
+    configuration_properties = _generate_mms_config_properties()
+
+    utils.write_file(MMS_CONFIG_FILE, configuration_properties)
+
+
+def _generate_mms_config_properties():
+    env = environment.Environment()
+
+    user_defined_configuration = {
+        'default_response_timeout': env.model_server_timeout,
+        'default_workers_per_model': env.model_server_workers,
+        'inference_address': 'http://0.0.0.0:{}'.format(env.http_port)
+    }
+
+    custom_configuration = str()
+
+    for key in user_defined_configuration:
+        value = user_defined_configuration.get(key)
+        if value:
+            custom_configuration += '{}={}\n'.format(key, value)
+
+    mms_default_configuration = utils.read_file(DEFAULT_MMS_CONFIG_FILE)
+
+    return mms_default_configuration + custom_configuration
+
+
+def _add_sigterm_handler(mms_process):
+    def _terminate(signo, frame):
+        try:
+            os.kill(mms_process.pid, signal.SIGTERM)
+        except OSError:
+            pass
+
+    signal.signal(signal.SIGTERM, _terminate)
+
+
+def _install_requirements():
+    logger.info('installing packages from requirements.txt...')
+    pip_install_cmd = [sys.executable, '-m', 'pip', 'install', '-r', REQUIREMENTS_PATH]
+
+    try:
+        subprocess.check_call(pip_install_cmd)
+    except subprocess.CalledProcessError:
+        logger.error('failed to install required packages, exiting')
+        raise ValueError('failed to install required packages')
+
+# retry for 10 seconds
+@retry(stop_max_delay=10 * 1000)
+def _retrieve_mms_server_process():
+    mms_server_processes = list()
+
+    for process in psutil.process_iter():
+        if MMS_NAMESPACE in process.cmdline():
+            mms_server_processes.append(process)
+
+    if not mms_server_processes:
+        raise Exception("mms model server was unsuccessfully started")
```

Comment:
Something is off with this message :)

Reply:
This was copied from mms_server code; I'm not really sure what this was supposed to catch either. I'll follow up with them during upstream merge